contracts/contracts/Umbra.sol

```diff
@@ -167,14 +168,24 @@ contract Umbra is Ownable {
    * @param _tokenAddr Address of the ERC20 token being withdrawn
    */
   function withdrawToken(address _acceptor, address _tokenAddr) external {
-    uint256 amount = tokenPayments[msg.sender][_tokenAddr];
-
-    require(amount > 0, "Umbra: No tokens available for withdrawal");
-
-    delete tokenPayments[msg.sender][_tokenAddr];
-    emit TokenWithdrawal(msg.sender, _acceptor, amount, _tokenAddr);
+    _withdrawTokenInternal(msg.sender, _acceptor, _tokenAddr, address(0), 0, UmbraHookable(0), "");
```

Comment:
So is your suggestion to basically re-implement/copy paste the withdrawal logic across the two methods? This is how it was previously, but I combined them in this PR because we now had 4 separate methods to withdraw from and it felt appropriate to consolidate the logic. That said, perhaps I'll explore 2 internal implementations as a gas saving technique as part of a gas-golfing pass.

Reply:
It was just a thought since this felt slightly dirty, though it does work fine. I don't feel strongly, so up to you which approach you want to use

Reply:
Cool, gonna leave for now, but will do some exploration while gas golfing later this week.